### DISCUSSION TITLE
Does hy have cons, car and cdr equivalents?

### DISCUSSION THREAD

**Discussion by yatsukha** [hylang/hy#2246] [Q&A] [answered]
I really miss having cons, car and cdr, but I'm fairly new so are there any equivalents?

**Comment by brandonwillard** [comment]
They were removed from Hy a while back, but I created a package that mostly reproduces their semantics: https://github.com/pythological/python-cons.

**Comment by Kodiologist** [comment, marked as answer]
To create a list, use `[ … ]`. To get the first item (a la `car`), use `(get x 0)`, or `(first x)` from  the `toolz` library. To get all but the first item (a la `cdr`), use ~`(slice x 1 None)`~ `(cut x 1 None)`, or `(rest x)` from Hyrule, or `(drop 1 x)` from `toolz`.

**Reply by Zambito1** [reply]
I am not super familiar with how slices work, but `cut` seems to behave closer to how I would expect for `cdr` to work.

```Lisp
=> (slice [1 2 3 4] 1 None)
(slice [1, 2, 3, 4] 1)
=> (slice '(a b c d) 1 None)
(slice hy.models.Expression([
  hy.models.Symbol('a'),
  hy.models.Symbol('b'),
  hy.models.Symbol('c'),
  hy.models.Symbol('d')]) 1)
=> (cut [1 2 3 4] 1 None)
[2 3 4]
=> (cut '(a b c d) 1 None)
'(b c d)
```

Is there a reason to prefer `slice` over `cut`?

**Reply by Kodiologist** [reply]
I really meant `cut`, yeah. `(cut x …)` is shorthand for `(get x (slice …))` and more often useful than `slice` itself, since it's rare to use a slice object for anything other than an immediate index operation.

**Reply by brandonwillard** [reply]
Just so everyone is aware, those functions are **not** equivalents to `cons`, `car`, and `cdr`; they're just list creation, `first`/`head`, and `rest`/`tail` functions.  They lack some important properties that would allow them to serve as replacements to `cons`, `car`, and `cdr` in most Lisp-like code.
